### DISCUSSION TITLE
Solara app using ipyleaflet disconnected after working some time.

### DISCUSSION THREAD

**Discussion by rururu** [widgetti/solara#291] [Q&A]
Dear solara users and team!

I have made an application using "ipyleaflet". Then I start inside it a process, that moves markers on a map. After some time the app hungs, its window become gray and apears a message: "! App is disconnected, and may not function properly. Please refresh the page." After refreshing the page, the app starts from the beginig, and I have to start the moving markers process again.
What con be a couse of this behavior? Thanks in advance for any help.

Sincerely,
  Ru

**Comment by maartenbreddels** [comment]
Hi Ruslan,

This is due to the WebSocket disconnecting. Are you using a cloud service to host your app? If so, which one?
We are working on this in https://github.com/widgetti/solara/pull/257 so I hope we can avoid this in the near future.

Regards,

Maarten

**Reply by rururu** [reply]
No, I am using this on local machine. Glad to hear that you are working on this issue.
